File: solutions/javascript/lasagna-master/1/lasagna-master.js
/// <reference path="./global.d.ts" />
// @ts-check

/**
 * Implement the functions needed to solve the exercise here.
 * Do not forget to export them so they are available for the
 * tests. Here an example of the syntax as reminder:
 *
 * export function yourFunction(...) {
 *   ...
 * }
 */
export function cookingStatus(minutes) {
  if (minutes == 0){
    return 'Lasagna is done.'
  }else if (minutes > 0){
    return 'Not done, please wait.'
  }else{
    return 'You forgot to set the timer.'
  }
}

export function preparationTime(layers = 0, prepTime = 2) {
  return layers.length * prepTime
}

export function quantities(layers) {
  const obj = {sauce: 0, noodles: 0}
  for (let layer of layers) {
    if (layer == 'sauce'){
      obj[layer] += 0.2
    }else if (layer == 'noodles'){
      obj[layer] += 50
    }else{
      continue
    }
  }
  console.log(obj)
  return obj
}

export function addSecretIngredient(friendsList, myList) {
  myList.push(friendsList[friendsList.length - 1])
}

export function scaleRecipe(recipe, amount) {
  const scaled = {};
  for (const key in recipe) {
    scaled[key] = recipe[key] * amount / 2;
  }
  return scaled;
}

File: solutions/go/lasagna-master/1/lasagna_master.go
package lasagna

// TODO: define the 'PreparationTime()' function
// DONE
func PreparationTime(layers []string, timeToPrep int) int {
    if timeToPrep == 0 {
        return len(layers) * 2
    }else{return len(layers) * timeToPrep} 
}

// TODO: define the 'Quantities()' function
// DONE
func Quantities(layers []string) (noodles int, sauce float64) {
    noodles = 0
    sauce = 0.0
    
    for i := 0; i < len(layers); i++ {
        if layers[i] == "noodles"{
            noodles += 50
        }else if layers[i] == "sauce"{
            sauce += 0.2
        }
    }
    return
}

// TODO: define the 'AddSecretIngredient()' function
// DONE
func AddSecretIngredient(friendsList []string, myList []string) {
    lastOfFriendsList := friendsList[len(friendsList)-1]
    myList[len(myList) - 1] = lastOfFriendsList 
}

// TODO: define the 'ScaleRecipe()' function
func ScaleRecipe(amounts []float64, portionsWanted int) []float64{
 	newQuantities := make([]float64, len(amounts))
    
	for i := 0; i < len(amounts); i++ {
		newQuantities[i] = amounts[i] * float64(portionsWanted) / 2
	}
    
	return newQuantities
}

// Your first steps could be to read through the tasks, and create
// these functions with their correct parameter lists and return types.
// The function body only needs to contain `panic("")`.
// 
// This will make the tests compile, but they will fail.
// You can then implement the function logic one by one and see
// an increasing number of tests passing as you implement more 
// functionality.
